ExampleMod/Content/NPCs/ExampleTravelingMerchant.cs

```diff
@@ -236,9 +248,37 @@ class ExampleTravelingMerchant : ModNPC
 		}
 
 		public override void LoadData(TagCompound tag) {
-			shopItems = tag.Get<List<Item>>("shopItems");
+  			shopItems.Clear();
+			shopItems.AddRange(tag.Get<List<Item>>("shopItems"));
 		}
 
+  		public override void SendExtraAI(BinaryWriter writer) {
+			if (!netShopItemsUpdate) {
```

Comment:
This is a good implementation of one of the ideas I had in mind, but it unfortunately has some downsides.
1. The information is not sent to new players immediately
2. The new player detection check can fail if a player leaves and joins in the same frame (unlikely but possible)
3. The update is sent to all players, not just the joining player, and also sent whenever a player leaves.

Given that `TravelingMerchantSystem` already exists, and is required for updating the spawn logic, you could send the items in the `NetSend` hook there.

Surprisingly, tModLoader doesn't offer any hooks in the `OnPlayerJoining` space (response to message 8). Despite vanilla having many systems which send packets with the same name, and an unused `IOnPlayerJoining` interface.

I like the idea of a more NPC centric approach, but I'd like something a bit more directed towards specific use cases. Note also that `SyncNPC` has some poorly implemented 'nearby NPC broadcast culling' via `NPC.netSkip`. This doesn't apply to `townNPC`, `boss` or `netAlways` NPCs, but people might not notice this particular limitation.

Reply:
Alright. See my two recent commits. The NPC is no longer responsible for saving/loading or syncing `shopItems`. Instead, the ModSystem handles that.